csrc/scheduler/matmul.cpp

```diff
@@ -734,6 +734,16 @@ void scheduleSplitKSum(
   splitk_sum->axis(-1)->parallelize(ParallelType::Vectorize);
 }
 
+bool needsTranposedLoad(TensorView* producer, TensorView* consumer) {
+  const auto map =
+      PairwiseRootDomainMap(producer, consumer).mapProducerToConsumer();
+  auto maybeProducerAlloc = producer->getMaybeAllocationDomain();
+  auto maybeConsumerRFactor = consumer->getMaybeRFactorDomain();
```

Comment:
nitpick: let's assert on `std::is_permutation`. Otherwise we can falsely claim needsTransposedLoad.